common/src/main/java/org/opensearch/ml/common/conversational/ConversationMeta.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.ml.common.conversational;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.search.SearchHit;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * Class for holding conversational metadata
+ */
+@AllArgsConstructor
+public final class ConversationMeta implements Writeable, ToXContentObject {
+
+    @Getter
+    private String id;
+    @Getter
+    private Instant created;
+    @Getter
+    private Instant lastHit;
+    @Getter
+    private int numInteractions;
+    @Getter
+    private String name;
```

Comment:
Do we need `name` field? I see you always using empty string as name in the examples https://github.com/opensearch-project/ml-commons/pull/1196#issuecomment-1674021347

> hmlin@Henrys-MBP OSContainer % curl localhost:9200/_plugins/ml/conversational/memory
{"conversations":[{"conversationId":"deGW4YkByJOQvDkDZ3Bm","createTime":"2023-08-10T22:34:37.788196552Z","lastInteractionTime":"2023-08-10T22:34:37.78820501Z","numInteractions":0,"name":"","user":null}]}

Reply:
The thought was that if you wanted the end user to be able to select between several ongoing conversations that they have going (think like the chatGPT web ui) then it would be helpful to have human-readable names associated with the conversation.